docfx_project/articles/bootstrapping-trust/part-01.encryption-everywhere.md

```diff
@@ -0,0 +1,310 @@
+private key, that prooves its unique identity. That private key is
+mathematically tied to a public key. The public key can be used to
+encrypt messages that only the private key holder can decrypt. Also, the
+public key cannot be used to derive the original private key. This
+functionality fits perfectly with what our distributed system needs!
+Alas, public-key cryptography introduces complex behaviors, setup , and
+management. In the next article, we will dive a little deeper into this
+topic. For now, let us take it on faith that it will serve our needs
+well.
+
+#### Setting It Up
+
+So we have decided that public-key cryptography is the answer. What does
+that mean? What do I have to do? Let us explore what would need to be
+done by a human or a piece of software automating this process. Don't
+worry if you don't get all of this; the gist is all you need for now.
+Later parts will expand upon this terminology. In fact, after reading
+the later parts, consider revisiting this part.
+
+Consider the following diagram of a "mesh" distributed system. This mesh
+could be any type of system such as a mesh of Ziti Routers, or maybe it
+is a system of sensors on an airplane. What they do does not matter.
+What matters is that this system has multiple pieces of software
+connecting amongst themselves. Consider what it means to accomplish this
+using public-key cryptography.
+
+![Image of a simple mesh](./images/mesh.png)
+
+In the diagram above, each system needs:
+
+- a key pair for client & server connections
+- to have the public keys of each system it is connecting to
+
+So what do we need to do? Drop into a CLI and start generating keys on
+each machine. Do that by using these commands:
+
+```
+openssl ecparam -name secp256k1-genkey -param_enc explicit -out private-key.pem
+```
+
+```
+openssl req -new -x509 -key private-key.pem -out server.pem -days 360
+```
+
+Voila - you now have a self-signed certificate (that no one trusts). You
+can repeat this for every piece of software in your mesh network, as it
+initially will be deployed. Preferably, you log into each machine and
+generate the private key there. Moving private keys on and off devices
+is a security risk and frowned upon. For maximum security, you would
+work with a
+[Hardware Security Module (HSM)](https://en.wikipedia.org/wiki/Hardware_security_module)
+such as those available embedded into devices or as a fob.
+
+Then you will need to copy each public certificate to every other
+machine and configure your software so that it trusts that certificate.
+The system will need to repeat this process any time the system adds a
+piece of software. If a machine is compromised, the analogous public
+certificate will need to be untrusted on every node in the mesh.
+
+Wow, this seems like quite a bit of work. Consider what this means when
+adding a node or removing one? Vising each machine and reconfiguring it
+each time is quite a bit of overhead. Let us see how Certificate
```

Comment:
this is the first time our newbie has read the term "Certificate Authority" and the next paragraph starts by telling the reader what they do. It would probably be a good idea to either give a cursory overview here or point them somewhere else to read up on it

Reply:
Softened it and linked to part 4.